modules/cockroachdb/cockroachdb.go

```diff
@@ -186,6 +186,12 @@ func addWaitingFor(req *testcontainers.GenericContainerRequest, opts options) er
 		}),
 	)
 
+	if req.WaitingFor == nil {
+		req.WaitingFor = defaultStrategy
+	} else {
+		req.WaitingFor = wait.ForAll(req.WaitingFor, defaultStrategy)
```

Comment:
Let's say that someone specified a waiting strategy of `wait.ForHTTP("/health").WithPort(defaultAdminPort),` similar to the default strategy, but this time they added a timeout. Would it override the default strategy here?

Reply:
It would get merged in meaning duplicate waits on the `wait.ForHTTP("/health")`

I agree this is cumbersome. I think what could be done is to pull the `wait.ForSQL` and then merge that into whatever the user is providing, this would get rid of the duplicate 

Overall it seems a bit awkward that connection logic is built into the wait for logic. Best approach might be to just separate that piece out entirely, however, I suspect there is some reason to keep it here though as the original author had some reason.

Reply:
@AhmedNSidd I separated out this connection logic and added a test based on your comment.